### DISCUSSION TITLE
How can I increase update times, and is control possible?

### DISCUSSION THREAD

**Discussion by apbarratt** [john30/ebusd#995] [Q&A]
Hi guys,

I'm afraid I'm very much struggling with configuring my setup for my heatpump.

My goals are to be able to control the heating and hot water in Home Assistant and also monitor its status and power usage too.

Thus far I've been able to get the current power usage in kW, the hot water temperature and a few other bitties like room temperature coming from the thermostat.  My biggest problem with these is it can be a great many minutes between updates.

I am able to see several "writable" values in Home Assistant, but I don't see anything obvious in terms of controlling behaviour of heating or hot water and some of the entities that show up are completely irrelevant.  Almost all entities are using technical names that not being a heat pump engineer are difficult for me to understand (a glossary somewhere would be nice 😅).  I also worry about some of the writable entities, for example I see an average outdoor temperature value, that is shown as a slider in Home Assistant and whilst I wish I had that control of the weather, it seems a tad unlikely :)

So far it's been useful in terms of being able to see when my hot water tank is low, but I still need to be at home to punch through the controls on the thermostat and tell it to boost the hot water.  This is sort of thing I'd very much like to configure so that my hot water gets a boost when I'm exporting surplus solar energy for instance.

My set up is this:
Vaillant Arotherm Plus heatpump
Vaillant sensoComfort thermostat
A single zone of central heating
A hotwater tank
A buffer tank in the loft
Two electronic boxes next to the buffer tank, I'll stick a quicky photo below as I'm not sure of exact model numbers.
Into the right hand box I've wired in my eBusd Adapter Version 5
A box on the wall outside also exists for the outside temperature.
My Home Assistant instance runs on a Raspberry Pi 4 and I've installed the eBusd Add On with it pointing at a clone of the config files latest directory.
I've updated my mqtt-hassio.cfg file with really only one change, `filter-direction = r|u|^w` to ensure I get the writable stuff showing up.

![IMG_1138](https://github.com/john30/ebusd/assets/642320/4d046892-13db-40f6-beac-8d83b58b7b1b)
![IMG_1137](https://github.com/john30/ebusd/assets/642320/92e0d667-4f61-48b0-ad4a-f1a95fa9c1f9)

When booting all this up, the auto discovered devices I'm getting in Home Assistant are `broadcast`, `basv` and `hmu`.  These seem to correspond to `latest/en/vaillant/15.basv.csv` and `latest/en/vaillant/08.hmu.csv` witht the broadcast one seeming to only show the value of the outside temperature.

`hmu` seems to have mostly information about the energy usage right now and sometimes seems to reflect modes when I do a hot water boost, but has no writable options.
`basv` seems to have far more information including room and outdoor temperatures, modes and statuses for all things including three central heating zones (as I say, I only have one) but again I can't figure out what steps I would take to, for instance, boost the hot water.
I have drop down menus for things like "rcmode" which provide options such as "off, thermostat or modulating".  I've no idea what modulating would mean.  "autoOff" modes include "night or eco".
Something called "sfMode" is especially alien to me, with modes including "auto, ventillation, load, veto, and party."  I can't say I can think of any situation in which a heating system requires a "party mode", but there we are :/
Auto, Day and Night modes seem to exist for many settings that mean nothing to me.
There are numerous options for cooling temperatures, but I don't have any cooling equipment (I wish I did, but alas I do not).

Many of the heating parts I think I may be able to tinker with and figure out and it's summer right now so my goals for this weekend are:

- Boost hot water on demand from Home Assistant
- Increase frequency of hot water temperature updates
  - Sometimes these can be half an hour old when someone has actually drained the tank
- Increase frequency of energy usage updates
  - I'm finding that I can turn on the heatpump and my energy cards can see the house using more but the heatpump isn't showing up as using any yet.  And the reverse is often true, with a hot water boost having finished but the heatpump's energy sensor still reporting that 2kW are being used for several minutes after the fact.

Any assistant would be very much appreciated :)
I also would love to contribute data where this could help improve things, so please do ask if there's anything I can provide in terms of logs etc as I see various "unknown codes" flitting by on the logs so perhaps I could help provide data there too.

**Comment by pm90pl** [comment]
> Increase frequency of energy usage updates
Same here, would like to know if there is any option of configurability in that area?

**Comment by pm90pl** [comment]
Ok, so time_t maxAge = 5*60; 5 minutes is the refresh frequency of the ebusd.
